### PR TITLE
TMPDIR patch for chocolate-doom

### PR DESCRIPTION
**Pull request by WMCB-Tech**
I just noticed that the tmpdir isn't patched yet, so it will spit out errors regarding with

`MIDI_LoadFile: Failed to open '/tmp/doom.mid'`

Setting `TEMP` environment variable should be work arounded for now if using mid files

Also first time doing a patch so hopefully things should go smoothly, guides are appreciated!

### CODE REVIEW

**Comment by WMCB-Tech** [comment]
Seems `git diff` fails, probably i should use `diff -uNr` instead according to Termux Packages wiki

File: packages/chocolate-doom/fix-tmpdir.patch [hunk 1]
```diff
@@ -0,0 +1,26 @@
+diff --git a/chocolate-doom-chocolate-doom-3.0.1/src/m_misc.c b/chocolate-doom-chocolate-doom-3.0.1.mod/src/m_misc.c
+index 7e69da8..d18cc0b 100644
+--- a/chocolate-doom-chocolate-doom-3.0.1/src/m_misc.c
```
Comment (suhan-paradkar) [inline packages/chocolate-doom/fix-tmpdir.patch]: Also.. you mostly need to chang line 3 and 4 to  
```
--- ./src/m_misc.c
```
And
```
+++ ./src/m_misc.c
```
.... That works for me usually after diff -uNr

File: packages/chocolate-doom/fix-tmpdir.patch [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+     // In Unix, just use /tmp.
+ 
+-    tempdir = "/tmp";
++    tempdir = "@TERMUX_PREFIX@/tmp";
+ #endif
+ 
+     return M_StringJoin(tempdir, DIR_SEPARATOR_S, s, NULL);
+diff --git a/chocolate-doom-chocolate-doom-3.0.1/src/setup/execute.c b/chocolate-doom-chocolate-doom-3.0.1.mod/src/setup/execute.c
+index 33472e7..1161c17 100644
+--- a/chocolate-doom-chocolate-doom-3.0.1/src/setup/execute.c
```
Comment (suhan-paradkar) [inline packages/chocolate-doom/fix-tmpdir.patch]: Even here...
```
--- ./src/setup/execute.c
```
And
```
+++ ./src/setup/execute.c
```

**Review by suhan-paradkar — CHANGES_REQUESTED** [review]

File: packages/chocolate-doom/fix-tmpdir.patch (hunk 1 above)
Comment (WMCB-Tech) [reply packages/chocolate-doom/fix-tmpdir.patch]: Thanks for the suggestion

File: packages/chocolate-doom/fix-tmpdir.patch (hunk 2 above)
Comment (suhan-paradkar) [reply packages/chocolate-doom/fix-tmpdir.patch]: It seems it works without that too

**Review by LeapofAzzam-backup — APPROVED** [review]

**Review by xeffyr — APPROVED** [review]

**Pull request merged**